taskipy/task_runner.py

```diff
@@ -17,11 +21,24 @@ def run(self, task_name: str, args: List[str]) -> int:
         for command in task.commands:
             if task.runner is not None:
                 command = f'{task.runner} {command}'
-
             process = subprocess.Popen(
                 command, shell=True, cwd=Path(self.__pyproject_path).parent
             )
 
+            def send_signal_to_task_process(signum: int, frame) -> None:  # pylint: disable=unused-argument
+                # pylint: disable=W0640
+                psutil_process_wrapper = psutil.Process(process.pid)
+
+                processes = psutil_process_wrapper.children(recursive=True)
```

Comment:
I think I understand why this didn't work before. The subprocess is actually a shell process, so passing a SIGTERM to it is not passed to the child process.

What we can do, is the following:
```suggestion
                sub_processes_of_taskipy_shell = psutil_process_wrapper.children()
                for process in sub_processes_of_taskipy_shell:
                    process.send_signal(signum)
```

Reply:
I think the process you called `sub_processes_of_taskipy_shell` here, is in fact the **shell itself**.
See the tree:
```
    6 pts/0    S+     0:01 /srv/worker/.venv/bin/python /srv/worker/.venv/bin/task worker
   10 pts/0    S+     0:00  \_ /bin/sh -c python -u main.py
   11 pts/0    Sl+    0:05      \_ python -u main.py
```

Should we keep recursive and loop over all children ? Or just send the signal to the **first none-sh subprocess** ?

Reply:
> ```
>   6 pts/0    S+     0:01 /srv/worker/.venv/bin/python /srv/worker/.venv/bin/task worker
>   10 pts/0    S+     0:00  \_ /bin/sh -c python -u main.py
>   11 pts/0    Sl+    0:05      \_ python -u main.py
>```

The first process is taskipy itself (where our code runs), so the subprocess should point at the second process (`/bin/sh -c ...`). So this method should work :)

Try altering the code to pass the signal to direct children. If your test passes, then I think that we'll be alright ;)

Reply:
Well I made a small change to pass the tests for both Linux and macOS. Are you ok with that ?